[CI] CliApiKeyIT testCliConnectionWithInvalidApiKey failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #11926 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/vinal7afy2pz2)
- [elasticsearch-periodic-platform-support #11926 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/vinal7afy2pz2)
- [elasticsearch-periodic-platform-support #11926 / windows-2025_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/6gtul7pkj34yy)
- [elasticsearch-periodic-platform-support #11926 / windows-2025_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/6gtul7pkj34yy)
- [elasticsearch-periodic-platform-support #11926 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/zqmawq36m67lo)
- [elasticsearch-periodic-platform-support #11926 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/zqmawq36m67lo)
- [elasticsearch-periodic-platform-support #11926 / windows-2025_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/gdrlh6wro7ahm)
- [elasticsearch-periodic-platform-support #11926 / windows-2025_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/gdrlh6wro7ahm)
- [elasticsearch-periodic-java-ea #193 / checkpart4_windows-2025](https://gradle-enterprise.elastic.co/s/jptlh2tnkat6s)
- [elasticsearch-periodic-java-ea #193 / checkpart4_windows-2025](https://gradle-enterprise.elastic.co/s/jptlh2tnkat6s)

**Reproduction Line:**
```
gradlew ":x-pack:plugin:sql:qa:server:security:with-ssl:javaRestTest" --tests "org.elasticsearch.xpack.sql.qa.security.CliApiKeyIT.testCliConnectionWithInvalidApiKey" -Dtests.seed=D9FA802C49375459 -Dtests.locale=bs-Latn -Dtests.timezone=Canada/Central -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.sql.qa.security.CliApiKeyIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCliConnectionWithInvalidApiKey'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: unconsumed lines
Expected: an empty collection
     but: <[	at org.elasticsearch.xpack.sql.client.JreHttpUrlConnection.request(JreHttpUrlConnection.java:190), 	at org.elasticsearch.xpack.sql.client.HttpClient.lambda$post$1(HttpClient.java:169), 	at org.elasticsearch.xpack.sql.client.JreHttpUrlConnection.http(JreHttpUrlConnection.java:82), 	at org.elasticsearch.xpack.sql.client.HttpClient.lambda$post$2(HttpClient.java:165), 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:74), 	at org.elasticsearch.xpack.sql.client.HttpClient.post(HttpClient.java:164), 	at org.elasticsearch.xpack.sql.client.HttpClient.query(HttpClient.java:131), 	at org.elasticsearch.xpack.sql.client.HttpClient.basicQuery(HttpClient.java:127), 	at org.elasticsearch.xpack.sql.cli.command.ServerQueryCliCommand.doHandle(ServerQueryCliCommand.java:30), 	at org.elasticsearch.xpack.sql.cli.command.AbstractServerCliCommand.handle(AbstractServerCliCommand.java:18), 	at 
[truncated]
```

**Issue Reasons:**
- [main] 6 consecutive failures in test testCliConnectionWithInvalidApiKey
- [main] 8 consecutive failures in step windows-2022_checkpart4_platform-support-windows
- [main] 8 consecutive failures in step windows-2025_checkpart4_platform-support-windows
- [main] 2 consecutive failures in step checkpart4_windows-2025
- [main] 36 failures in test testCliConnectionWithInvalidApiKey (4.1% fail rate in 870 executions)
- [main] 8 failures in step windows-2022_checkpart4_platform-support-windows (100.0% fail rate in 8 executions)
- [main] 8 failures in step windows-2025_checkpart4_platform-support-windows (100.0% fail rate in 8 executions)
- [main] 2 failures in step checkpart4_windows-2025 (100.0% fail rate in 2 executions)
- [main] 4 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 4 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-java-ea (66.7% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/docs/changelog/143408.yaml
+++ b/docs/changelog/143408.yaml
@@ -1,0 +1,7 @@
+area: SQL
+issues:
+ - 143018
+ - 143019
+pr: 143408
+summary: Fix SQL client parsing of array header values
+type: bug

--- a/x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/RemoteFailure.java
+++ b/x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/RemoteFailure.java
@@ -269,10 +269,20 @@ public class RemoteFailure {
             }
             String name = parser.getText();
             token = parser.nextToken();
-            if (token != JsonToken.VALUE_STRING) {
+            String value;
+            if (token == JsonToken.VALUE_STRING) {
+                value = parser.getText();
+            } else if (token == JsonToken.START_ARRAY) {
+                List<String> values = new ArrayList<>();
+                while ((token = parser.nextToken()) != JsonToken.END_ARRAY) {
+                    if (token == JsonToken.VALUE_STRING) {
+                        values.add(parser.getText());
+                    }
+                }
+                value = String.join(", ", values);
+            } else {
                 throw new IOException("expected header value but was [" + token + "][" + parser.getText() + "]");
             }
-            String value = parser.getText();
             headers.put(name, value);
         }
 

--- a/x-pack/plugin/sql/sql-client/src/test/java/org/elasticsearch/xpack/sql/client/RemoteFailureTests.java
+++ b/x-pack/plugin/sql/sql-client/src/test/java/org/elasticsearch/xpack/sql/client/RemoteFailureTests.java
@@ -64,6 +64,18 @@ public class RemoteFailureTests extends ESTestCase {
         assertEquals(singletonMap("WWW-Authenticate", "Basic realm=\"security\", charset=\"UTF-8\""), failure.headers());
     }
 
+    public void testParseMissingAuthMultipleSchemes() throws IOException {
+        RemoteFailure failure = parse("missing_auth_multiple_schemes.json");
+        assertEquals("security_exception", failure.type());
+        assertEquals(
+            "unable to authenticate with provided credentials and anonymous access is not allowed for this request",
+            failure.reason()
+        );
+        assertThat(failure.remoteTrace(), containsString("AuthenticationService.authenticate"));
+        assertNull(failure.cause());
+        assertEquals(singletonMap("WWW-Authenticate", "Basic realm=\"security\", charset=\"UTF-8\", ApiKey"), failure.headers());
+    }
+
     public void testNoError() {
         IOException e = expectThrows(IOException.class, () -> parse("no_error.json"));
         assertEquals(

--- a/x-pack/plugin/sql/sql-client/src/test/resources/remote_failure/missing_auth_multiple_schemes.json
+++ b/x-pack/plugin/sql/sql-client/src/test/resources/remote_failure/missing_auth_multiple_schemes.json
@@ -1,0 +1,23 @@
+{
+  "error" : {
+    "root_cause" : [
+      {
+        "type" : "security_exception",
+        "reason" : "unable to authenticate with provided credentials and anonymous access is not allowed for this request",
+        "additional_unsuccessful_credentials" : "API key: Illegal base64 character 5f",
+        "header" : {
+          "WWW-Authenticate" : ["Basic realm=\"security\", charset=\"UTF-8\"", "ApiKey"]
+        },
+        "stack_trace" : "ElasticsearchSecurityException[unable to authenticate]\n\tat org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(AuthenticationService.java:99)"
+      }
+    ],
+    "type" : "security_exception",
+    "reason" : "unable to authenticate with provided credentials and anonymous access is not allowed for this request",
+    "additional_unsuccessful_credentials" : "API key: Illegal base64 character 5f",
+    "header" : {
+      "WWW-Authenticate" : ["Basic realm=\"security\", charset=\"UTF-8\"", "ApiKey"]
+    },
+    "stack_trace" : "ElasticsearchSecurityException[unable to authenticate]\n\tat org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(AuthenticationService.java:99)"
+  },
+  "status" : 401
+}

